File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int>res(2);
        int i=0;
        int j=0;
        int n=nums.size();
        for(i=0;i<n;i++)
        {
            for(j=0;j<n;j++)
            {
                if(i==j)
                {
                    continue;
                }
                if(nums[i]+nums[j]==target)
                {
                    res[0]=i;
                    res[1]=j;
                    return res;
                }
            }
        }
        return res;
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        my_dict={}
        for i in range(len(nums)):
            if target-nums[i] in my_dict:
                return [my_dict[target-nums[i]],i]
            else:
                my_dict[nums[i]]=i
        
            








        # for i in range(len(nums)):
        #     for j in range(i+1,len(nums)):
        #         if nums[i]+nums[j]==target:
        #             return [i,j]
        
        # my_dict={}
        # for i in range(len(nums)):
        #     if target-nums[i] in my_dict:
        #         return [my_dict[target-nums[i]],i]
        #     my_dict[nums[i]]=i